src/wp-includes/fonts.php

```diff
@@ -131,7 +131,44 @@ function wp_get_font_dir() {
 	 *
 	 * @param array $defaults The original fonts directory data.
 	 */
-	return apply_filters( 'font_dir', $defaults );
+	$font_dir = apply_filters( 'font_dir', $default_dir );
+
+	wp_mkdir_p( $font_dir['path'] );
+
+	if ( ! is_dir( $font_dir['path'] ) || ! wp_is_writable( $font_dir['path'] ) ) {
+
+		// If the default directory was filtered it uses the filtered values and doesnt fall back to the wp-content/uploads/fonts.
+		if ( $font_dir !== $default_dir ) {
+			$font_dir['error'] = sprintf(
+				/* translators: %s: Directory path. */
+				__( 'Unable to create directory %s. Is its parent directory writable by the server?' ),
+				esc_html( str_replace( ABSPATH, '', $font_dir['path'] ) )
+			);
+			return $font_dir;
+		}
+
+		$upload_dir = wp_get_upload_dir();
+		$font_dir   = array(
+			'path'    => path_join( $upload_dir['basedir'], 'fonts' ),
+			'url'     => $upload_dir['baseurl'] . '/fonts',
+			'subdir'  => '',
+			'basedir' => path_join( $upload_dir['basedir'], 'fonts' ),
+			'baseurl' => $upload_dir['baseurl'] . '/fonts',
+			'error'   => false,
+		);
+
```

Comment:
As @peterwilsoncc mentions above thinking it would probably be good to run the `font_dir` filter here again as the values were reset (hardcoded).

Reply:
I'd really prefer to just run it once after WordPress figures out what it will present as the default.

Reply:
If I'm understanding correctly you prefer this function to do something like this:
```
if ( is_dir( WP_CONTENT_DIR ) && is_writable( WP_CONTENT_DIR ) ) {
    $default_font_dir = WP_CONTENT_DIR . `/fonts`;
elseif ( is_dir( WP_CONTENT_DIR . '/uploads' ) && is_writable( WP_CONTENT_DIR . '/uploads' ) ) {
    // This will only work if /uploads already exists. The assumption is that /uploads cannot be created as WP_CONTENT_DIR either doesn't exist or is not writable.
    $default_font_dir = WP_CONTENT_DIR . `/uploads/fonts`;
}

$default_font_dir = apply_filters( 'font_dir', $default_font_dir );

// Try to crate the fonts dir
```